lib/domain/usecases/permission_usecases/handle_location_permission.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:dartz/dartz.dart';
+import 'package:injectable/injectable.dart';
+import 'package:movna/domain/entities/system_permission.dart';
+import 'package:movna/domain/failures.dart';
+import 'package:movna/domain/usecases/base_usecases.dart';
+import 'package:movna/domain/usecases/permission_usecases/get_location_permission.dart';
+import 'package:movna/domain/usecases/permission_usecases/request_location_permission.dart';
+
+/// Retrieves the [SystemPermissionStatus] of the location permission.
+///
+/// If this status is other than [SystemPermissionStatus.permanentlyDenied],
+/// shows a dialog for the user to grant the permission and returns the
+/// resulting permission status.
+///
+/// If the initial status is [SystemPermissionStatus.permanentlyDenied] or a
+/// granted status to some degree, this usecase only returns the status and does
+/// nothing else.
+@injectable
+class HandleLocationPermission
```

Comment:
`Handle` is way too generic, and it is usually a good choice to name variables and classes differently. Maybe `Request` or `Ensure` there ?

Reply:
i know handle is generic but this use-case does not necessarily requests the permission, it does so only if the permission is not granted and not permanently denied

So neither `ensure` not `request` works well